lib/macho/macho_file.rb

```diff
@@ -189,7 +189,38 @@ def add_command(lc)
       set_sizeofcmds(new_sizeofcmds)
       @raw_data[header.class.bytesize + sizeofcmds, cmd_raw.bytesize] = cmd_raw
 
-      # synchronize fields with the raw data
+      synchronize_fields! if synchronize
+    end
+
+    # Delete a load command from the Mach-O.
+    # @param lc [MachO::LoadCommand] the load command being deleted
+    # @param synchronize [Boolean] whether or not to synchronize the instance fields
+    # @return [void]
+    # @note This is public, but methods like {#delete_rpath} should be preferred.
+    #  Setting `synchronize` to false **will break the instance** unless
+    #  {#synchronize_fields!} is called **immediately** afterwards.
+    def delete_command(lc, synchronize = true)
+      @raw_data.slice!(lc.view.offset, lc.cmdsize)
+
+      # decrease the number of load commands by the number deleted
+      set_ncmds(ncmds - 1)
+
+      # decrease the total size of commands by the size of the deleted commands
+      set_sizeofcmds(sizeofcmds - lc.cmdsize)
+
+      # pad the space after the load commands to preserve offsets
+      null_pad = "\x00" * lc.cmdsize
+      @raw_data.insert(header.class.bytesize + sizeofcmds, null_pad)
+
+      synchronize_fields! if synchronize
+    end
+
+    # Synchronize the instance's fields with the raw Mach-O data.
+    # @return [void]
+    # @note This method is public, but should (almost) never need to be called.
+    #  The exception to this rule is when methods like {#add_command} and
+    #  #{delete_command} have been called with `synchronization_false`.
+    def synchronize_fields!
```

Comment:
> When I'm thinking of synchronization, I'm thinking more about flushing the changes that are only in memory to disk (or in this case to raw data), i.e. exactly in the opposite direction.

That makes sense, especially considering that the only other bang-method in `MachOFile` is `write!`. I'll remove the `!`.

I went with `synchronize_fields` because I ended up using it in the constructor(s) as well (where I think it would sound a little strange to call `reload_fields` or something similar). That being said, `reload_fields` or perhaps `refresh_fields` would be fine by me if we revert to populating `@header` and `@load_commands` explicitly in the constructors.